src/nvim/viml/executor/vim.lua

```diff
@@ -0,0 +1,2296 @@
+                                             idx1, idx1_position,
+                                             idx2, idx2_position, lock)
+    local idx1, idx2 = list.get_slice_indicies(
+      state, lst.length, IDX_UNLET, list.get_index,
+      idx1, idx1_position,
+      idx2, idx2_position
+    )
+    if idx1 == nil then
+      return nil
+    end
+    local already_set = {}
+    for i = idx1,idx2 do
+      lst[locks_idx][i] = lock
+      local v = lst[i]
+      vim_type(v).set_container_lock(state, depth, v, lock, already_set)
+    end
+    return true
+  end,
+-- {{{4 Operators support
+  cmp_priority = 10,
+  cmp = function(state, ic, eq, val1, val1_position, val2, val2_position)
+    if not (is_list(val1) and is_list(val2)) then
+      return err.err(state, val2_position, true,
+                     'E691: Can only compare List with List')
+    elseif not eq then
+      return err.err(state, val1_position, true,
+                     'E692: Invalid operation for Lists')
+    end
+    return bisimilar_tables(ic, val1, val2, false) and 0 or 1
+  end,
+  num_op_priority = 3,
+  do_add = function(state, copy_ret, val1, val1_position, val2, val2_position)
+    if not (is_list(val1) and is_list(val2)) then
+      -- Error out
+      if is_list(val1) then
+        return list.as_number(state, val1, val1_position)
+      else
+        return list.as_number(state, val2, val2_position)
+      end
+    end
+    if not copy_ret then
+      local t = vim_type(val1)
+      if not list.check_container_lock(state, t, val1, val1_position, nil) then
+        return nil
+      end
+    end
+    local length1 = val1.length
+    local ret = copy_ret and copy_table(val1) or val1
+    for i, v in ipairs(val2) do
+      ret[length1 + i] = v
+    end
+    ret.length = length1 + val2.length
+    return ret
+  end,
+  add = function(state, ...)
+    return list.do_add(state, true, ...)
+  end,
+  mod_add = function(state, ...)
+    return list.do_add(state, false, ...)
+  end,
+-- }}}4
+})
+
+-- {{{3 Dictionary
+dict = join_tables(container, {
+-- {{{4 New
+  type_number = BASE_TYPE_DICTIONARY,
+  can_be_self = true,
+  new = add_type_table(function(state, ...)
+    if select('#', ...) % 2 ~= 0 then
+      return err.err(state, position, true,
+                     'Internal error: odd number of arguments to dict:new')
+    end
+    local ret = {[length_idx]=select('#', ...)/2, [locks_idx]={}}
+    for i = 1,ret[length_idx] do
+      local key = select(2*i - 1, ...)
+      local val = select(2*i, ...)
+      if not (key and val) then
+        return nil
+      end
+      ret[get_string(state, key)] = val
+    end
+    return ret
+  end),
+-- {{{4 Modification support
+-- {{{4 Assignment support
+  assign_subscript = function(state, dct, dct_position, key, key_position, val)
+    if key == '' then
+      return err.err(state, key_position, true,
+                     'E713: Cannot use empty key for dictionary')
```

Comment:
This was changed in https://github.com/vim/vim/commit/0921ecff1c5a74541bad6c073e8ade32247403d8